test/optimizer/timestamp_comparisons.test

```diff
@@ -0,0 +1,116 @@
+# name: test/optimizer/timestamp_comparisons.test
+# description: timestampt comparisons should be correct
+# group: [optimizer]
+
+require icu
+
+statement ok
+create table t (t timestamptz);
+
+statement ok
```

Comment:
Set the calendar to `gregorian` too. Otherwise people running the tests in some areas (like China) may get strange results.